be/test/storage/utils_test.cpp

```diff
@@ -41,4 +43,16 @@ TEST_F(TestUtils, test_valid_decimal) {
     ASSERT_TRUE(valid_decimal("31.4", 3, 1));
     ASSERT_TRUE(valid_decimal("314.15925", 8, 5));
 }
+
+TEST_F(TestUtils, test_is_tracker_hit_hard_limit) {
+    std::unique_ptr<MemTracker> tracker = std::make_unique<MemTracker>(1000, "test", nullptr);
+    tracker->consume(2000);
+    ASSERT_TRUE(is_tracker_hit_hard_limit(tracker.get(), 30, 30));
```

Comment:
what if the soft limit is larger than hard limit? what does that mean?

Reply:
```
int64_t limit_ratio = std::max(hard_limit_percent * 100 / soft_limit_percent, 100);
```
If the soft limit is larger than hard limit, hard limit is treat as same as soft limit.

Reply:
looks like "soft" is harder?

Reply:
Why?

Reply:
ok

> ```
> int64_t limit_ratio = std::max(hard_limit_percent * 100 / soft_limit_percent, 100);
> ```
> 
> If the soft limit is larger than hard limit, hard limit is treat as same as soft limit.

ok. got it. this sentence confuses me. it is the soft limit set to the same as hard limit. However, it doesn't matter because eventually the calculation will result a 100 ratio limit.